### PR TITLE
Add types for mem-fs

### PR DESCRIPTION
**Pull request by wrumsby**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/mem-fs/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// Definitions by: My Food Bag <https://github.com/MyFoodBag>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import events = require('events');
+import stream = require('stream');
+import File = require('vinyl');
+
+export function create(...args: any[]): create.Store;
+
+export namespace create {
```
Comment (wrumsby) [inline types/mem-fs/index.d.ts]: TODO: rename `namespace` to `memFs`.

File: types/mem-fs/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Type definitions for mem-fs 1.1
+// Project: https://github.com/sboudrias/mem-fs#readme
+// Definitions by: My Food Bag <https://github.com/MyFoodBag>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import events = require('events');
```
Comment (wrumsby) [inline types/mem-fs/index.d.ts]: Add

```
/// <reference types="node" />
```

File: types/mem-fs/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+export namespace memFs {
+    interface Store extends EventEmitter {
+        add: (file: File, content: string) => void;
+        each: (callback: (file: File, index: number) => void) => void;
+        get: (filepath: string) => File;
+        stream: () => Transform;
+    }
+
+    const prototype: {
+    };
```
Comment (plantain-00) [inline types/mem-fs/index.d.ts]: It seems the `prototype` can be removed

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Comment by wrumsby** [comment]
@plantain-00 `prototype` removed.

File: types/mem-fs/index.d.ts (hunk 3 above)
Comment (minestarks) [reply types/mem-fs/index.d.ts]: @wrumsby you can go ahead and remove this line.

File: types/mem-fs/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Type definitions for mem-fs 1.1
+// Project: https://github.com/sboudrias/mem-fs#readme
```
Comment (minestarks) [inline types/mem-fs/index.d.ts]: minor point, but project link without the `#readme` is the convention.

**Review by minestarks — CHANGES_REQUESTED** [review]

**Comment by wrumsby** [comment]
@plantain-00 apologies, I hadn't actually saved my file locally - updated now.

**Comment by wrumsby** [comment]
@minestarks `#readme` removed from project URL.

**Review by minestarks — APPROVED** [review]

**Pull request merged**